var/spack/repos/builtin/packages/lfortran/package.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Lfortran(CMakePackage):
+    """Modern interactive LLVM-based Fortran compiler"""
+
+    homepage = "https://lfortran.org"
+    url = "https://lfortran.github.io/tarballs/release/lfortran-0.19.0.tar.gz"
+    git = "https://github.com/lfortran/lfortran.git"
+    maintainers = ["certik"]
+
+    # The build process uses 'git describe --tags' to get the package version
+    version("main", branch="main", get_full_repo=True)
+    version("0.19.0", sha256="d496f61d7133b624deb3562677c0cbf98e747262babd4ac010dbd3ab4303d805")
+
+    variant("llvm", default=True, description="Build with LLVM support")
+    variant("stacktrace", default=True, description="Build with stacktrace support")
+
+    depends_on("python@3:", type="build", when="@main")
+    depends_on("cmake", type="build")
+    depends_on("llvm@11:15", type="build", when="+llvm")
```

Comment:
Since LFortran depends on `clang` to compile Fortran code, the "run" dependency type should be added for LLVM.
```suggestion
    depends_on("llvm@11:15", type=("build", "run"), when="+llvm")
```